### PR TITLE
[PLUGIN-1792] Enable macro for plugin config script

### PR DESCRIPTION
**Pull request by psainics**
## Enable macro for plugin config script

Jira : [PLUGIN-1792](https://cdap.atlassian.net/browse/PLUGIN-1792)

### Description

This PR makes the plugin config `script` a macro.

- Demo
image

### Code change

- Modified `PythonEvaluator.java`

### CODE REVIEW

File: src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java [hunk 1]
```diff
@@ -229,7 +230,7 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     }
     // try evaluating the script to fail application creation
     // if the script is invalid (only possible for interpreted mode)
-    if (config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
+    if (config.executionMode != null && config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
```
Comment (itsankit-google) [inline src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java]: please add a check here that script is not macro before we try to compile it in `JythonPythonExecutor`

**Review by itsankit-google — CHANGES_REQUESTED** [review]
Please add a unit test

File: src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java (hunk 1 above)
Comment (psainics) [reply src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java]:
```java
if (config.executionMode != null &&
    !config.containsMacro(Config.SCRIPT) &&
    config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) 
```

**Comment by psainics** [comment]
> Please add a unit test

Is there api for testing macro ?

File: src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java [hunk 2]
```diff
@@ -229,7 +230,9 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     }
     // try evaluating the script to fail application creation
     // if the script is invalid (only possible for interpreted mode)
-    if (config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
+    if (config.executionMode != null &&
```
Comment (itsankit-google) [inline src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java]: `!containsMacro(config.executionMode)`

We should also throw validation exception if `executionMode` is `null`:
```
private static final String EXECUTION_MODE = "executionMode";
if (!containsMacro(config.EXECUTION_MODE) && Strings.isNullOrEmpty(config.executionMode)) {
.....
return;
}

if (!config.containsMacro(config.EXECUTION_MODE) && !config.containsMacro(Config.SCRIPT) &&
        config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
      try {
        new JythonPythonExecutor(config).initialize(null);
      } catch (IOException | InterruptedException e) {
        collector.addFailure("Could not compile script: " + e.getMessage(), null)
          .withStacktrace(e.getStackTrace()).withConfigProperty(Config.SCRIPT);
      }
    }
```

Comment (psainics) [reply src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java]: Updated !

**Review by itsankit-google — APPROVED** [review]

**Pull request merged**